include/hc_printf.hpp

```diff
@@ -64,40 +76,34 @@ enum PrintfError {
 
 static inline PrintfPacket* createPrintfBuffer(hc::accelerator& a, const unsigned int numElements) {
   PrintfPacket* printfBuffer = NULL;
-  if (numElements > 6) {
+  if (numElements > 5) {
```

Comment:
5 looks to be a magic value worthwhile be enumed.

also the logic would return a `nullptr`, but other APIs such as `deletePrintBuffer` doesn't guard `nullptr`. I would recommend you improve the test case to check `nullptr` cases as well.

Reply:
Should test case also check for error cases from printf call on gpu?

Reply:
since the test case use FileCheck to check its output perhaps it's fine just to add nullptr check to make sure the tests won't crash, but could capture the case when the implementation returns a nullptr.

Reply:
Do you mean use an assertion for the case where create returns a nullptr? Or another sort of capture?

Reply:
since you use FileCheck to check the output i think it's not good to let the test case crash.

Reply:
I've committed a change for the test cases to check for nullptr. Just a note, printf, processprintf, and deleteprintf should be guarding against nullptr buffer. So, test case will fail by `CHECK-DAG`, since the test won't crash if createprintf fails.